cql3/functions/aggregate_function.hh

```diff
@@ -31,6 +31,13 @@ public:
      */
     virtual std::unique_ptr<aggregate> new_aggregate() = 0;
 
+    /**
+     * Creates a <code>Aggregate Function</code> that can be reduced.
+     *
+     * @return a reducible <code>Aggregate Function</code>.
+     */
+    virtual ::shared_ptr<aggregate_function> reducible_aggregate_function() = 0;
+
```

Comment:
Please make the comments helpful by explaining the differences. How does an aggregate signal that it is not reducible? By returning null here?

Reply:
Ah, I see is_reducible below. Please indicate this is only legal if is_reducible().

Reply:
Added to function's comment